proxyclient/utils.py

```diff
@@ -94,6 +94,21 @@ def inst(self, op):
         if cnt:
             raise ProxyError("Exception occurred")
 
+    def compressed_writemem(self, dest, data, progress):
+        if not len(data):
+            return
+
+        payload = lzma.compress(data)
+        compressed_size = len(payload)
+        compressed_addr = self.malloc(compressed_size)
+
+        self.iface.writemem(compressed_addr, payload, progress)
+        decompressed_size = self.proxy.xzdec(compressed_addr, compressed_size, dest, len(data))
+
+        assert decompressed_size == len(data)
+
+        self.free(compressed_addr)
```

Comment:
Maybe try/finally guard this? Remember that malloc/free are Python-side, so e.g. if a proxy command fails and breaks m1n1 free will still work.

Reply:
good idea, done.

Reply:
Nice, I like it!